src/main/java/com/example/commandtest/service/CommandProcessor.java

```diff
@@ -0,0 +1,87 @@
+    private final Map<String, Consumer<String>> commandHandlers = new HashMap<>();
+
+    public CommandProcessor(UniversityService universityService) {
+        this.universityService = universityService;
+        initializeCommands();
+    }
+
+    private void initializeCommands() {
+        commandHandlers.put("Who is head of department ", this::handleWhoIsHeadCommand);
+        commandHandlers.put("Show department ", this::handleStatisticCommand);
+        commandHandlers.put("Show the average salary for the department ", this::handleAverageSalaryCommand);
+        commandHandlers.put("Show count of employee for ", this::handleEmployeeCountCommand);
+        commandHandlers.put("Global search by ", this::handleGlobalSearchCommand);
+    }
+
+    public void startListening() {
+        try (Scanner scanner = new Scanner(System.in)) {
+            log.info("Welcome! Write command (or 'exit' for exit):");
+            while (true) {
+                String userInput = scanner.nextLine();
+                if ("exit".equals(userInput)) {
+                    log.info("Exit from application.");
+                    break;
+                }
+                processUserInput(userInput);
+            }
+        }
+    }
+
+    private void processUserInput(String userInput) {
+        commandHandlers.forEach((command, handler) -> {
+            if (userInput.startsWith(command)) {
+                executeCommand(handler, userInput.substring(command.length()).trim());
+            }
+        });
+    }
+
+    private void executeCommand(Consumer<String> commandAction, String argument) {
+        try {
+            commandAction.accept(argument);
+        } catch (EntityNotFoundException e) {
+            log.warn("Entity not found for command: {}, error: {}", argument, e.getMessage());
+        } catch (Exception e) {
+            log.error("An error occurred while executing command: {}", argument, e);
+        }
+    }
+
+    private void handleWhoIsHeadCommand(String departmentName) {
+        System.out.println(universityService.getDepartmentHead(departmentName));
+    }
+
+    private void handleStatisticCommand(String departmentName) {
+        System.out.println(universityService.getStatistic(departmentName));
+    }
+
+    private void handleAverageSalaryCommand(String departmentName) {
+        System.out.println("The average salary for the department " + departmentName + " is " +
+                universityService.getAverageSalary(departmentName));
+    }
+
+    private void handleEmployeeCountCommand(String departmentName) {
+        System.out.println("The count of employee for " + departmentName + " is " +
+                universityService.getCountOfEmployee(departmentName));
+    }
+
+    private void handleGlobalSearchCommand(String template) {
+        String searchResults = universityService.searchEverywhere(template);
+        System.out.println(searchResults.isEmpty() ? "No matches found." : searchResults);
+    }
```

Comment:
Imagine having 10000 commands

it will be better to 
```suggestion
private void handle(Command command) {
    command.doSomeShit()
}

Reply:
Это происки Сатаны